### PR TITLE
Add ORT fused adam optimizer

### PR DESCRIPTION
**Pull request by JingyaHuang**
# What does this PR do?

* Integrate ort fused adam optimizer
* Subclass TrainingArguments to include ort optimizers
* Update training examples
* Add to the nightly test

-------------------------
__Who can review__

@zhijxu-MS @carzh

### CODE REVIEW

**Comment by JingyaHuang** [comment]
> @JingyaHuang let's discuss here for ort-fused-adam.
> 
> with your new changes, i can run it locally, such as "question-answering/run_qa.py".
> 
> the [issue ](https://github.com/huggingface/optimum/pull/229#issuecomment-1181914030) you encountered looks like you are using an old ort, could you install a newer one. the install command is `pip uninstall -y onnxruntime-training && pip install --pre --no-deps --no-index onnxruntime-training -f https://download.onnxruntime.ai/onnxruntime_nightly_cu113.html && TORCH_CUDA_ARCH_LIST='7.0;8.0+PTX' python -m torch_ort.configure`

Hi @zhijxu-MS,  thanks for the advice. I tested with onnxruntime nightly and it works well for me. 👍 

As the previous branch was a little messy, I closed the previous and opened another PR here. Also I updated the examples as well as the test of `ORTTrainer`. Can you take a look if everything is ok for you? Thx!

File: optimum/onnxruntime/trainer.py [hunk 1]
```diff
@@ -1426,3 +1424,82 @@ def _wrap_model(self, model, training=True, dataloader=None):
...
+            "eps": args.adam_epsilon,
+        }
+        if args.optim == ORTOptimizerNames.ADAMW_ORT_FUSED:
+            try:
+                from onnxruntime.training.optim import FusedAdam
+
+                optimizer_cls = FusedAdam
+                optimizer_kwargs.update(adam_kwargs)
+            except ImportError:
+                raise ValueError(
```
Comment (regisss) [inline optimum/onnxruntime/trainer.py:1500]: It should be `raise ImportError` no?

**Review by regisss — APPROVED** [review]
LGTM @JingyaHuang!!

File: optimum/onnxruntime/trainer.py (hunk 1 above)
Comment (JingyaHuang) [reply optimum/onnxruntime/trainer.py:1500]: Good catch!

**Comment by zhijxu-MS** [comment]
> > @JingyaHuang let's discuss here for ort-fused-adam.
> > with your new changes, i can run it locally, such as "question-answering/run_qa.py".
> > the [issue ](https://github.com/huggingface/optimum/pull/229#issuecomment-1181914030) you encountered looks like you are using an old ort, could you install a newer one. the install command is `pip uninstall -y onnxruntime-training && pip install --pre --no-deps --no-index onnxruntime-training -f https://download.onnxruntime.ai/onnxruntime_nightly_cu113.html && TORCH_CUDA_ARCH_LIST='7.0;8.0+PTX' python -m torch_ort.configure`
> 
> Hi @zhijxu-MS, thanks for the advice. I tested with onnxruntime nightly and it works well for me. 👍
> 
> As the previous branch was a little messy, I closed the previous and opened another PR here. Also I updated the examples as well as the test of `ORTTrainer`. Can you take a look if everything is ok for you? Thx!

i tested 4 optimum examples locally, they can be run w/ or w/o ort-fused-adam. and make sure ort-fused-adam is called by adding log.

**Review by mfuntowicz — APPROVED** [review]
LGTM 😃

**Comment by JingyaHuang** [comment]
Thanks for the review @mfuntowicz @regisss @zhijxu-MS , will merge it.

**Pull request merged**